src/Microsoft.TestPlatform.TestHostProvider/Microsoft.TestPlatform.TestHostProvider.csproj

```diff
@@ -26,7 +26,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="$(MicrosoftExtensionsDependencyModelPackageVersion)" />
-    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.2" />
+    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.2" Condition="'$(OS)' == 'Windows_NT'" />
```

Comment:
This one is interesting.
/cc: @mayankbansal018

Reply:
Yeah, it seems odd that netcoreapp2.1 Windows fails without it but netcoreapp2.1 Linux doesn't need it.

Reply:
Are we going with the OS check then ?

Reply:
@crummel Can you please reply on this